tests/test_preprocessing/test_match_columns.py

```diff
@@ -126,23 +128,51 @@ def test_drop_columns_when_more_columns_in_test_than_train(df_vartypes, df_na):
     assert match_columns.fill_value is np.nan
     assert match_columns.verbose is True
     assert match_columns.missing_values == "ignore"
+    assert match_columns.match_dtypes is False
     # test fit attrs
     assert list(match_columns.feature_names_in_) == list(train.columns)
     assert match_columns.n_features_in_ == 4
     # test transform output
     pd.testing.assert_frame_equal(expected_result, transformed_df)
 
```

Comment:
Are we testing that the transformer raises an error when match_dtypes is not a boolean? If not, we need to introduce one.